src/python/pants/engine/target.py

```diff
@@ -1020,7 +1008,7 @@ def compute_value(cls, raw_value: Optional[Any], *, address: Address) -> Optiona
         return value_or_default
 
 
-class BoolField(PrimitiveField, metaclass=ABCMeta):
+class TriBoolField(PrimitiveField[Optional[bool]], metaclass=ABCMeta):
```

Comment:
I'm not sure why `BoolField` wasn't using `ScalarField` in the first place. Might have been oversight. Do you think it would work here?

Reply:
That should totally work!